pkgs/by-name/di/direwolf-unstable/package.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  fetchFromGitHub,
+  direwolf,
+  espeak,
+  nix-update-script,
+  hamlibSupport ? true,
+  gpsdSupport ? true,
+  extraScripts ? false,
+}:
+
+(direwolf.override {
+  inherit hamlibSupport gpsdSupport extraScripts;
+}).overrideAttrs
+  (finalAttrs: {
+    # Next version is 1.8
+    version = "1.8-unstable-2025-04-29";
```

```suggestion
    version = "1.7-unstable-2025-04-29";
```
Please review the [package versioning guidelines](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning):
- If we're tracking a commit from a repository without a version assigned, then the version attribute should be the **latest upstream version** preceding that commit, followed by `-unstable-` and the date of the fetched commit. The date must be in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) "YYYY-MM-DD" format.

That's to say, if 1.8 isn't tagged, it can't be added here.

---

Huh, I missed that. I'll fix `gpredict-unstable`'s version in a seperate PR then.

---

I'm not sure we should be adding unstable package versions to nixpkgs at all - this seems like a NUR thing.

---

Both have very slow release cycles and have some significant features that I would like to use that aren't present in the latest stable version. This shouldn't require much maintenance, considering updates are going to be, for the most part, completely automated. If there is a policy or other concrete reason against including unstable variants of packages, then fair enough, but both are, in my opinion, value added.

I do get what you mean about it maybe being a better fit for NUR. In fact, I do have a custom direwolf derivation in my own nur-packages. I feel that that shouldn't preclude it from being in nixpkgs though. Nixpkgs has a higher standard for quality than NUR does, and I believe this meets that higher standard.